src/screens/courses/list/TrainerCourses/index.tsx

```diff
@@ -16,16 +16,31 @@ import { NextSlotsStepType } from '../../../../types/StepTypes';
 import { RootBottomTabParamList, RootStackParamList } from '../../../../types/NavigationType';
 import { useGetLoggedUserId } from '../../../../store/main/hooks';
 import commonStyles from '../../../../styles/common';
-import { BLENDED, OPERATIONS, TRAINER } from '../../../../core/data/constants';
+import { BLENDED, COMPLETED, FORTHCOMING, OPERATIONS, TRAINER } from '../../../../core/data/constants';
 import styles from '../styles';
-import { isInProgress, isForthcoming, isCompleted, getElearningSteps, formatNextSteps } from '../helper';
+import { getElearningSteps, formatNextSteps, getCourseStatus } from '../helper';
 import { CourseDisplayType } from '../types';
 import TrainerEmptyState from '../TrainerEmptyState';
 
 const formatCoursesDiplaysContent = (courses: BlendedCourseType[]) => {
-  const coursesInProgress = courses.filter(c => isInProgress(c));
-  const forthcomingCourses = courses.filter(c => isForthcoming(c));
-  const completedCourses = courses.filter(c => isCompleted(c));
+  const coursesInProgress = [];
+  const forthcomingCourses = [];
+  const completedCourses = [];
+
+  for (let i = 1; i < courses.length; i += 1) {
```

Comment:
Pourquoi passer par l'index ? Tu peux directement parcourir les elements de `courses` non ?

Reply:
oui on peut faire un foreach ,j'avais mis une boucle for car je crois que c'est légèrement plus efficace mais c'est pas perceptible dans ce cas là j'ai l'impression

Reply:
Et tu peux pas faire `const course of courses` ?

Reply:
Et un for const course of ....` classique ne fonctionne pas ?